Thanks, I didn't modify any android code, but I can give you some clues


According to https://github.com/Expensify/react-native/actions/runs/5080321413/jobs/9137029909, the CI error message is
```
/app/sdks/hermes/API/hermes/hermes.cpp:639:32: error: non-virtual member function marked 'override' hides virtual member functions
         const jsi::Value &value) override;
                                  ^
/app/ReactCommon/jsi/jsi/jsi.h:340:3: note: hidden overloaded virtual function 'facebook::jsi::Runtime::setPropertyValue' declared here: type mismatch at 1st parameter ('facebook::jsi::Object &' vs 'const jsi::Object &')
```

The reason for the error here is because `facebook::jsi::Object` should be `const facebook::jsi::Object`

So we can integrate this PR https://github.com/facebook/react-native/commit/03b17d9af7e4e3ad3f9ec078b76d0ffa33a3290e#diff-4d691d151cba9dc2f79cb3ae582fc99ab8d639dda355d367aef25fc136ad0f93R340, but I guess that will also take some time, so we can merge our PR first, because It doesn't modify any android code

--- a/packages/react-native/Libraries/Text/TextInput/Multiline/RCTUITextView.h
+++ b/packages/react-native/Libraries/Text/TextInput/Multiline/RCTUITextView.h
@@ -24,6 +24,7 @@ NS_ASSUME_NONNULL_BEGIN
 
 @property (nonatomic, assign) BOOL contextMenuHidden;
 @property (nonatomic, assign, readonly) BOOL textWasPasted;
+@property (nonatomic, assign, readonly) BOOL dictationRecognizing;
 @property (nonatomic, copy, nullable) NSString *placeholder;
 @property (nonatomic, strong, nullable) UIColor *placeholderColor;
 

--- a/packages/react-native/Libraries/Text/TextInput/Multiline/RCTUITextView.m
+++ b/packages/react-native/Libraries/Text/TextInput/Multiline/RCTUITextView.m
@@ -261,6 +261,19 @@ static UIColor *defaultPlaceholderColor()
   return [super canPerformAction:action withSender:sender];
 }
 
+#pragma mark - Dictation
+
+- (void)dictationRecordingDidEnd
+{
+  _dictationRecognizing = YES;
+}
+
+- (void)removeDictationResultPlaceholder:(id)placeholder willInsertResult:(BOOL)willInsertResult
+{
+  [super removeDictationResultPlaceholder:placeholder willInsertResult:willInsertResult];
+  _dictationRecognizing = NO;
+}
+
 #pragma mark - Placeholder
 
 - (void)_invalidatePlaceholderVisibility

--- a/packages/react-native/Libraries/Text/TextInput/RCTBackedTextInputViewProtocol.h
+++ b/packages/react-native/Libraries/Text/TextInput/RCTBackedTextInputViewProtocol.h
@@ -18,6 +18,7 @@ NS_ASSUME_NONNULL_BEGIN
 @property (nonatomic, copy, nullable) NSString *placeholder;
 @property (nonatomic, strong, nullable) UIColor *placeholderColor;
 @property (nonatomic, assign, readonly) BOOL textWasPasted;
+@property (nonatomic, assign, readonly) BOOL dictationRecognizing;
 @property (nonatomic, assign) UIEdgeInsets textContainerInset;
 @property (nonatomic, strong, nullable) UIView *inputAccessoryView;
 @property (nonatomic, strong, nullable) UIView *inputView;

--- a/packages/react-native/Libraries/Text/TextInput/RCTBaseTextInputView.m
+++ b/packages/react-native/Libraries/Text/TextInput/RCTBaseTextInputView.m
@@ -119,8 +119,12 @@ RCT_NOT_IMPLEMENTED(-(instancetype)initWithFrame : (CGRect)frame)
                      }
                    }];
 
-  BOOL shouldFallbackToBareTextComparison =
-      [self.backedTextInputView.textInputMode.primaryLanguage isEqualToString:@"dictation"] ||
+  BOOL shouldFallbackDictation = [self.backedTextInputView.textInputMode.primaryLanguage isEqualToString:@"dictation"];
+  if (@available(iOS 16.0, *)) {
+    shouldFallbackDictation = self.backedTextInputView.dictationRecognizing;
+  }
+
+  BOOL shouldFallbackToBareTextComparison = shouldFallbackDictation ||
       [self.backedTextInputView.textInputMode.primaryLanguage isEqualToString:@"ko-KR"] ||
       self.backedTextInputView.markedTextRange || self.backedTextInputView.isSecureTextEntry ||
       fontHasBeenUpdatedBySystem;

--- a/packages/react-native/Libraries/Text/TextInput/Singleline/RCTUITextField.h
+++ b/packages/react-native/Libraries/Text/TextInput/Singleline/RCTUITextField.h
@@ -24,6 +24,7 @@ NS_ASSUME_NONNULL_BEGIN
 @property (nonatomic, assign) BOOL caretHidden;
 @property (nonatomic, assign) BOOL contextMenuHidden;
 @property (nonatomic, assign, readonly) BOOL textWasPasted;
+@property (nonatomic, assign, readonly) BOOL dictationRecognizing;
 @property (nonatomic, strong, nullable) UIColor *placeholderColor;
 @property (nonatomic, assign) UIEdgeInsets textContainerInset;
 @property (nonatomic, assign, getter=isEditable) BOOL editable;

--- a/packages/react-native/Libraries/Text/TextInput/Singleline/RCTUITextField.m
+++ b/packages/react-native/Libraries/Text/TextInput/Singleline/RCTUITextField.m
@@ -142,6 +142,19 @@
   return [super canPerformAction:action withSender:sender];
 }
 
+#pragma mark - Dictation
+
+- (void)dictationRecordingDidEnd
+{
+  _dictationRecognizing = YES;
+}
+
+- (void)removeDictationResultPlaceholder:(id)placeholder willInsertResult:(BOOL)willInsertResult
+{
+  [super removeDictationResultPlaceholder:placeholder willInsertResult:willInsertResult];
+  _dictationRecognizing = NO;
+}
+
 #pragma mark - Caret Manipulation
 
 - (CGRect)caretRectForPosition:(UITextPosition *)position

